From a9c2b316154e97cc8f7f8945caf33638c35591de Mon Sep 17 00:00:00 2001
From: Sebastiano Barezzi <barezzisebastiano@gmail.com>
Date: Tue, 13 Dec 2022 19:36:08 +0100
Subject: [PATCH 006/157] Revert "arm64: dts: vendor: qcom: Remove unused
 conflicting disp_rdump_memory region"

* Xiaomi uses this reserved region

This reverts commit 9191aae299627fd9730818de6a7348c413553786.

Change-Id: Ie0d8a002ad216ddf886057630f0e630164ad80a5
---
 arch/arm64/boot/dts/vendor/qcom/bengal-iot-low-ram.dtsi  | 2 ++
 arch/arm64/boot/dts/vendor/qcom/bengal-low-ram.dtsi      | 2 ++
 arch/arm64/boot/dts/vendor/qcom/bengal.dtsi              | 5 +++++
 arch/arm64/boot/dts/vendor/qcom/bengalp-iot-low-ram.dtsi | 2 ++
 arch/arm64/boot/dts/vendor/qcom/khaje.dtsi               | 5 +++++
 arch/arm64/boot/dts/vendor/qcom/kona.dtsi                | 5 +++++
 arch/arm64/boot/dts/vendor/qcom/lagoon.dtsi              | 5 +++++
 arch/arm64/boot/dts/vendor/qcom/lito.dtsi                | 5 +++++
 arch/arm64/boot/dts/vendor/qcom/scuba.dtsi               | 5 +++++
 9 files changed, 36 insertions(+)

diff --git a/arch/arm64/boot/dts/vendor/qcom/bengal-iot-low-ram.dtsi b/arch/arm64/boot/dts/vendor/qcom/bengal-iot-low-ram.dtsi
index 52304bfee9da..ab4a6ae57e5b 100644
--- a/arch/arm64/boot/dts/vendor/qcom/bengal-iot-low-ram.dtsi
+++ b/arch/arm64/boot/dts/vendor/qcom/bengal-iot-low-ram.dtsi
@@ -23,6 +23,8 @@
 
 /delete-node/ &secure_display_memory;
 
+/delete-node/ &disp_rdump_memory;
+
 &reserved_memory {
 	hyp_mem: hyp_region@45700000 {
 		compatible = "removed-dma-pool";
diff --git a/arch/arm64/boot/dts/vendor/qcom/bengal-low-ram.dtsi b/arch/arm64/boot/dts/vendor/qcom/bengal-low-ram.dtsi
index 584b45035d15..05c53c52643c 100644
--- a/arch/arm64/boot/dts/vendor/qcom/bengal-low-ram.dtsi
+++ b/arch/arm64/boot/dts/vendor/qcom/bengal-low-ram.dtsi
@@ -24,6 +24,8 @@
 
 /delete-node/ &secure_display_memory;
 
+/delete-node/ &disp_rdump_memory;
+
 &reserved_memory {
 	hyp_mem: hyp_region@45700000 {
 		compatible = "removed-dma-pool";
diff --git a/arch/arm64/boot/dts/vendor/qcom/bengal.dtsi b/arch/arm64/boot/dts/vendor/qcom/bengal.dtsi
index 8512e9cc5b79..519586abfd05 100644
--- a/arch/arm64/boot/dts/vendor/qcom/bengal.dtsi
+++ b/arch/arm64/boot/dts/vendor/qcom/bengal.dtsi
@@ -409,6 +409,11 @@
 			label = "cont_splash_region";
 		};
 
+		disp_rdump_memory: disp_rdump_region@5c000000 {
+			reg = <0x0 0x5c000000 0x0 0x00f00000>;
+			label = "disp_rdump_region";
+		};
+
 		dfps_data_memory: dfps_data_region@5cf00000 {
 			reg = <0x0 0x5cf00000 0x0 0x0100000>;
 			label = "dfps_data_region";
diff --git a/arch/arm64/boot/dts/vendor/qcom/bengalp-iot-low-ram.dtsi b/arch/arm64/boot/dts/vendor/qcom/bengalp-iot-low-ram.dtsi
index 428ba6a83df9..7451dc1c66ec 100644
--- a/arch/arm64/boot/dts/vendor/qcom/bengalp-iot-low-ram.dtsi
+++ b/arch/arm64/boot/dts/vendor/qcom/bengalp-iot-low-ram.dtsi
@@ -23,6 +23,8 @@
 
 /delete-node/ &secure_display_memory;
 
+/delete-node/ &disp_rdump_memory;
+
 &reserved_memory {
 	hyp_mem: hyp_region@45700000 {
 		compatible = "removed-dma-pool";
diff --git a/arch/arm64/boot/dts/vendor/qcom/khaje.dtsi b/arch/arm64/boot/dts/vendor/qcom/khaje.dtsi
index 7858e1c5f40c..d5e31f809aef 100644
--- a/arch/arm64/boot/dts/vendor/qcom/khaje.dtsi
+++ b/arch/arm64/boot/dts/vendor/qcom/khaje.dtsi
@@ -410,6 +410,11 @@
 			label = "cont_splash_region";
 		};
 
+		disp_rdump_memory: disp_rdump_region@5c000000 {
+			reg = <0x0 0x5c000000 0x0 0x00f00000>;
+			label = "disp_rdump_region";
+		};
+
 		dfps_data_memory: dfps_data_region@5cf00000 {
 			reg = <0x0 0x5cf00000 0x0 0x0100000>;
 			label = "dfps_data_region";
diff --git a/arch/arm64/boot/dts/vendor/qcom/kona.dtsi b/arch/arm64/boot/dts/vendor/qcom/kona.dtsi
index 7601bab953ef..86300eda2df7 100644
--- a/arch/arm64/boot/dts/vendor/qcom/kona.dtsi
+++ b/arch/arm64/boot/dts/vendor/qcom/kona.dtsi
@@ -519,6 +519,11 @@
 			label = "cont_splash_region";
 		};
 
+		disp_rdump_memory: disp_rdump_region@9c000000 {
+			reg = <0x0 0x9c000000 0x0 0x00800000>;
+			label = "disp_rdump_region";
+		};
+
 		dfps_data_memory: dfps_data_region@9e300000 {
 			reg = <0x0 0x9e300000 0x0 0x0100000>;
 			label = "dfps_data_region";
diff --git a/arch/arm64/boot/dts/vendor/qcom/lagoon.dtsi b/arch/arm64/boot/dts/vendor/qcom/lagoon.dtsi
index 6d8b26e946a5..b13324d100dc 100644
--- a/arch/arm64/boot/dts/vendor/qcom/lagoon.dtsi
+++ b/arch/arm64/boot/dts/vendor/qcom/lagoon.dtsi
@@ -460,6 +460,11 @@
 			label = "cont_splash_region";
 		};
 
+		disp_rdump_memory: disp_rdump_region@0xa0000000 {
+			reg = <0x0 0xA0000000 0x0 0x02300000>;
+			label = "disp_rdump_region";
+		};
+
 		dfps_data_memory: dfps_data_region {
 			reg = <0x0 0xA2300000 0x0 0x0100000>;
 			label = "dfps_data_region";
diff --git a/arch/arm64/boot/dts/vendor/qcom/lito.dtsi b/arch/arm64/boot/dts/vendor/qcom/lito.dtsi
index c4c5dd0d3e10..452e680871d7 100644
--- a/arch/arm64/boot/dts/vendor/qcom/lito.dtsi
+++ b/arch/arm64/boot/dts/vendor/qcom/lito.dtsi
@@ -509,6 +509,11 @@
 			label = "dfps_data_region";
 		};
 
+		disp_rdump_memory: disp_rdump_region@0xa0000000 {
+			reg = <0x0 0xA0000000 0x0 0x02300000>;
+			label = "disp_rdump_region";
+		};
+
 		dump_mem: mem_dump_region {
 			compatible = "shared-dma-pool";
 			alloc-ranges = <0x0 0x00000000 0x0 0xffffffff>;
diff --git a/arch/arm64/boot/dts/vendor/qcom/scuba.dtsi b/arch/arm64/boot/dts/vendor/qcom/scuba.dtsi
index c52d68b1bfc4..f1857653e9ae 100644
--- a/arch/arm64/boot/dts/vendor/qcom/scuba.dtsi
+++ b/arch/arm64/boot/dts/vendor/qcom/scuba.dtsi
@@ -291,6 +291,11 @@
 			label = "dfps_data_region";
 		};
 
+		disp_rdump_memory: disp_rdump_region@5c000000 {
+			reg = <0x0 0x5c000000 0x0 0x00f00000>;
+			label = "disp_rdump_region";
+		};
+
 		qseecom_mem: qseecom_region {
 			compatible = "shared-dma-pool";
 			alloc-ranges = <0x0 0x00000000 0x0 0xffffffff>;
-- 
2.48.1

